src/commands/random_shiggy.rs

```diff
@@ -0,0 +1,40 @@
+use crate::api::guzzle::REQWEST_CLIENT;
+use reqwest::StatusCode;
+use serde::Deserialize;
+use serenity::builder::CreateApplicationCommand;
+use serenity::model::prelude::application_command::CommandDataOption;
+
+const URL: &str = "https://safebooru.donmai.us/posts/random.json?tags=kemomimi-chan_(naga_u)+naga_u&only=file_url";
+const ERROR_MSG: &str = "couldn't get a shiggy";
+
+#[derive(Deserialize)]
+struct SafebooruResponse {
+	file_url: String,
+}
+
+pub async fn run(_: &[CommandDataOption]) -> String {
+	let resp = match REQWEST_CLIENT
+		.execute(REQWEST_CLIENT.get(URL).build().unwrap())
+		.await
+	{
+		Ok(r) => r,
+		Err(e) => return format!("{} ({:?})", ERROR_MSG, e),
+	};
+
+	match resp.status() {
+		StatusCode::OK => match resp.json::<SafebooruResponse>().await {
+			Ok(sr) => sr.file_url,
+			Err(e) => format!("{} ({:?})", ERROR_MSG, e),
+		},
+		other => {
+			println!("{}", resp.text().await.unwrap());
+			format!("{} ({:?})", ERROR_MSG, other)
+		}
+	}
+}
```

Comment:
could this api logic be moved to something like `src/api/shiggy.rs`? i prefer to keep all of the api stuff there, while keeping things for the commands themselves in `commands/`. you can see [src/api/guzzle.rs](https://github.com/uku3lig/teawieBot/blob/random-shiggy/src/api/guzzle.rs) and [src/commands/random_teawie.rs](https://github.com/uku3lig/teawieBot/blob/random-shiggy/src/commands/random_teawie.rs#L6) for an example of what i mean